nmrfx-analyst-gui/src/main/java/org/nmrfx/processor/gui/ProcessorController.java

```diff
@@ -231,40 +228,48 @@ public String toString() {
     private AtomicBoolean needToFireEvent = new AtomicBoolean(false);
     private AtomicReference<Dataset> saveObject = new AtomicReference<>();
     ScheduledFuture futureUpdate = null;
+    private boolean viewIfPossible = true;
 
 
-    public static ProcessorController create(FXMLController fxmlController, StackPane processorPane, PolyChart chart) {
-        Fxml.Builder builder = Fxml.load(ProcessorController.class, "ProcessorScene.fxml")
-                .withParent(processorPane);
+    public static ProcessorController create(FXMLController fxmlController, NmrControlRightSidePane nmrControlRightSidePane, PolyChart chart) {
+        Fxml.Builder builder = Fxml.load(ProcessorController.class, "ProcessorScene.fxml");
         ProcessorController controller = builder.getController();
 
         controller.chart = chart;
         chart.setProcessorController(controller);
         controller.chartProcessor.setChart(chart);
         controller.chartProcessor.setFxmlController(fxmlController);
-        controller.processorPane = processorPane;
-        controller.pane = builder.getNode();
-        fxmlController.processorCreated(controller.pane);
-
+        controller.nmrControlRightSidePane = nmrControlRightSidePane;
+        fxmlController.processorCreated(controller.mainBox);
+        controller.show();
         return controller;
     }
 
+    public Pane getPane() {
+        return mainBox;
+    }
+
     public void show() {
-        if (processorPane.getChildren().isEmpty()) {
-            processorPane.getChildren().add(pane);
-            pane.setVisible(true);
+        if (!nmrControlRightSidePane.getChildren().contains(mainBox)) {
+            nmrControlRightSidePane.getChildren().clear();
+            nmrControlRightSidePane.getChildren().add(mainBox);
         }
```

Comment:
Can this code be moved to NmrCotnrolRightSidePane? Since we already have isContentShowing(), we could have a displayContent() or showContent() or something like that?

Reply:
And ideally remove these show/hide methods in each content if we can, and let the side pane manage showing/hiding.

Reply:
Ah I think we can move the lines 253-255, I don't know if I can remove the show/hide methods from Processor Controller, because of wanting to keep track if the user had the processor controller open when they switch between charts in multi chart display, if we remove it then they will have to reclick the processing button, if they switch the active chart to something without a processor controller and then switch back, not sure how often someone would do that though

Reply:
actually moved the viewIfPossible boolean into FXMLController since it was more related to the status of the toggle buttons than the individual controllers so was able to completely remove the hide/show methods